Erin dev

erinyoung opened the pull request:
This is my solution to 

https://github.com/StaPH-B/docker-builds/issues/147

I didn't update the main readme because I saw that there was an active pull request.

kapsakcj commented on parallel/20200722/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+
+# metadata
+LABEL base.image="ubuntu:xenial"
+LABEL version="1"
```
`LABEL version="1"` --> `LABEL dockerfile.version="1"`

erinyoung replied:
I'll get that right eventually

kapsakcj commented on parallel/20200722/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+LABEL version="1"
+LABEL software="GNU Parallel"
+LABEL software.version="20200722"
+LABEL container.version="1"
```
Please delete the container.version LABEL since that info is captured (more coherently) with dockerfile.version

kapsakcj commented on parallel/20200722/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+                     libxml-parser-perl \
+                     libperlio-utf8-strict-perl \
+                     libmoose-perl \
+                     libgraph-perl
```
Let's make this docker image as tiny/agile as possible. You could add this to the end of the apt-get steps:
```Dockerfile
[...]
libmoose-perl \
libgraph-perl && \
rm -rf /var/lib/apt/lists/* && apt-get autoclean
```

erinyoung replied:
Sounds good

kapsakcj commented:
What would you like to name this docker image as? `staphb/parallel`? `staphb/parallel-perl`? 

OK with just the default latest tag? Any additional tags you'd like? `staphb/<image-name>:latest`?

erinyoung commented:
I like staphb/parallel-perl better, since it basically has functioning perl and parallel. I think I would need to change the directory name to reflect that.

Or should I remove all the perl? 

I don't really use it, I just kept it in since it seemed like everyone else was using it.

kapsakcj commented:
You already have the dockerfile, I'd say let's keep it. No need to remove the perl stuff. Can't hurt to have this available.

Please do adjust the directory name to `parallel-perl` so that we can keep it consistent with the name of the dockerhub repo

erinyoung commented:
I don't think it needs any additional tags.

erinyoung commented:
Do you think I can update the main readme?

kapsakcj commented:
I think the readme on your branch is slightly out of date since I just updated the readme on master branch this AM.

You can update the readme here before we merge the PR, or we can do it afterwards and commit directly to master branch. The latter might be a little bit simpler. Up to you.

Thanks for making those changes I requested!

erinyoung commented:
I just added to my pull request, because I didn't notice that this wasn't closed, yet. I added a new mafft container. The people at mafft have added some new features that make aligning SARS-CoV-2 less painful.

I'll update the readme once this has been merged.

kapsakcj commented:
Hey Erin, I'll try to review your changes this afternoon, pending on how long my appt at the car dealership takes. If not I'll look at this first thing on Monday morning. Thanks for tagging me, I forgot we had this PR open 😬

kapsakcj commented:
I made some changes to the mafft dockerfile, to reduce the size by a little bit, added an ARG variable, and updated a metadata label.
```
erin/mafft                   7.475-slim         1c309dbfb6df   11 seconds ago   109MB
erin/mafft                   7.475              63bef0c28daf   15 minutes ago   147MB
```
Built successfully after making the changes and I could pull up help options. I don't think mafft uses any perl, so hopefully we can get away without adding that locale variable in there too.

kapsakcj commented:
made one small addition to the parallel-perl dockerfile, so that it would build successfully. 

Docs for this and mafft look good. Thanks so much for adding those!

I'll merge these and push to dockerhub shortly!
